src/Mod/Path/Path/Post/scripts/dynapath_4060_post.py

```diff
@@ -0,0 +1,608 @@
+    if OUTPUT_COMMENTS:
+        gcode += linenumber() + "(T)" + "BEGIN PREAMBLE$\n"
+    for line in PREAMBLE.splitlines(True):
+        gcode += linenumber() + line
+    gcode += linenumber() + UNITS + "\n"
+
+    for obj in objectslist:
+
+        # Skip inactive operations
+        if hasattr(obj, "Active"):
+            if not obj.Active:
+                continue
+        if hasattr(obj, "Base") and hasattr(obj.Base, "Active"):
+            if not obj.Base.Active:
+                continue
+        if hasattr(obj, "ClearanceHeight"):
+            clearanceHeight = obj.ClearanceHeight.Value
+
+        # Fix fixture Offset label...
+        if obj.Label in GCODE_MAP:
+            obj.Label = GCODE_MAP[obj.Label]
+
+        # do the pre_op
+        if OUTPUT_COMMENTS:
+            gcode += (
+                linenumber()
+                + "(T)"
+                + str.upper("begin operation: " + obj.Label)
+                + "$\n"
+            )
+            gcode += (
+                linenumber()
+                + "(T)"
+                + str.upper("machine units: ")
+                + "%s/%s$\n"
+                % (str.upper(UNIT_SPEED_FORMAT[0:2]), str.upper(UNIT_SPEED_FORMAT[3:6]))
+            )
+
+        # get coolant mode
+        coolantMode = "None"
+        if (
+            hasattr(obj, "CoolantMode")
+            or hasattr(obj, "Base")
+            and hasattr(obj.Base, "CoolantMode")
+        ):
+            if hasattr(obj, "CoolantMode"):
+                coolantMode = obj.CoolantMode
+            else:
+                coolantMode = obj.Base.CoolantMode
+
+        # turn coolant on if required
+        if OUTPUT_COMMENTS:
+            if not coolantMode == "None":
+                gcode += (
+                    linenumber()
+                    + "(T)"
+                    + str.upper("Coolant On:" + coolantMode)
+                    + "$\n"
+                )
+        if coolantMode == "Flood":
+            gcode += linenumber() + "M8" + "\n"
+        # if coolantMode == "Mist":
+        #    gcode += linenumber() + "M7" + "\n"
+
+        # process the operation gcode
+        gcode += parse(obj)
+
+        # do the post_op
+        if OUTPUT_COMMENTS:
+            gcode += (
+                linenumber()
+                + "(T)"
+                + str.upper("finish operation: " + obj.Label)
+                + "$\n"
+            )
+
+        # turn coolant off if required
+        if not coolantMode == "None":
+            if OUTPUT_COMMENTS:
+                gcode += (
+                    linenumber()
+                    + "(T)"
+                    + str.upper("Coolant Off:" + coolantMode)
+                    + "$\n"
+                )
+            gcode += linenumber() + "M9" + "\n"
+
+    # do the post_amble
+    if OUTPUT_COMMENTS:
+        gcode += linenumber() + "(T)" + "BEGIN POSTAMBLE$\n"
+    for line in POSTAMBLE.splitlines(True):
+        gcode += linenumber() + line
+    gcode += "E\n"
+
```

Comment:
Should there be an "E" in that line?

Reply:
If the E is an End of program, why not include it in the postamble?